src/main/java/org/edumips64/core/CacheSimulator.java

```diff
@@ -41,251 +47,76 @@ public CacheMemory getL1InstructionCache() {
         return L1InstructionCache;
     }
 
+    // Cache type enumeration
     public enum CacheType {
         L1_DATA,
         L1_INSTRUCTION,
         L1_UNIFIED
     }
 
-
-    public static class CacheConfig {
-        public int size, blockSize, associativity, penalty;
-
+    // Backward compatibility alias for CacheConfig
```

Comment:
Look, there is *no* need for any backward compatibility. This is a self-contained application. Please rewrite your change to do what was asked in the original issue without any concerns for backwards compatibility. I expect the tests to work and that's it, no more efforts on backwards compatibility.